jscomp/runtime/js_set.ml

```diff
@@ -1,3 +1,34 @@
 (** ES6 Set API *)
 
 type 'a t
+
+external make : unit -> 'a t = "Set" [@@mel.new]
+external fromArray : 'a array -> 'a t = "Set" [@@mel.new]
+external toArray : 'a t -> 'a array = "Array.from"
+external size : 'a t -> int = "size" [@@mel.get]
+external add : value:'a -> 'a t = "add" [@@mel.send.pipe: 'a t]
```

Comment:
I'm not sure returning `'a t` is a good idea. I guess you did it to have a nice pipe API, but on the JS side the set is actually mutated, and this API doesn't make it obvious. See https://github.com/melange-re/melange/pull/1061/files#diff-64db361f42339ab2d557daeaff34549e0d79fc297dfc2411317c6d64f2e6557bR33-R41

Reply:
Indeed, but the documentation mentions the return value is the set indeed. So we'd be lying by pretending it's `unit` even if that makes it more obvious that the set was mutated, no? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/add#return_value